### PR TITLE
Add jnp.matrix_transpose() and jax.Array.mT

### PR DESCRIPTION
**Pull request by jakevdp**
This is an API proposed by the Python Array API Standard (https://data-apis.org/array-api/2022.12/). It's lightweight enough that there's hardly any downside to supporting it in JAX.

Related to #16099

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -538,6 +538,28 @@ def transpose(a: ArrayLike, axes: Optional[Sequence[int]] = None) -> Array:
   return lax.transpose(a, axes_)
 
 
+@util._wraps(getattr(np, 'matrix_transpose', None))
+def matrix_transpose(x: ArrayLike, /) -> Array:
+  """Transpose the last two dimensions of x.
```
Comment (hawkinsp) [inline jax/_src/numpy/lax_numpy.py:543]: Nit: "Transposes"

**Review by hawkinsp — APPROVED** [review]
Consider replacing the `_T` helper used in the following two places:
* https://github.com/google/jax/blob/e464dc870054c86bb6fee4ecca2ebf066f32ade5/jax/_src/numpy/linalg.py#L36
* https://github.com/google/jax/blob/e464dc870054c86bb6fee4ecca2ebf066f32ade5/jax/_src/lax/linalg.py#L381

?

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py]: Done

**Comment by jakevdp** [comment]
Thanks - I added a second commit using `mT` in the linalg code.

**Review by hawkinsp — APPROVED** [review]

**Pull request merged**